submissions/yaripey/frogger/app.js

```diff
@@ -0,0 +1,124 @@
+// Tile size
+const tile = {
+    width: 101,
+    height: 85
+}
+
+// Enemies our player must avoid
+const Enemy = function (x, y) {
+    // Variables applied to each of our instances go here,
+    // we've provided one for you to get started
+    this.x = x;
+    this.y = y;
+    this.changeSpeed();
+    // The image/sprite for our enemies, this uses
+    // a helper we've provided to easily load images
+    this.sprite = 'images/enemy-bug.png';
```

Comment:
this.x, this.y, this.sprite - properties that can be generalized into a base class. For example, the class 'Character' can be a base class, which is extended by Enemy and Player classes. And you can use inheritance for them both.

Class extension can be implemented using Subclass.prototype = Object.create(Superclass.prototype);
[Here is useful article](https://ui.dev/javascript-inheritance-and-the-prototype-chain/) about functional inheritance.